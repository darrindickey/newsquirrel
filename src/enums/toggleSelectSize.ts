enum toggleSelectSize {
  SMALL,
  LARGE
}

export default toggleSelectSize;