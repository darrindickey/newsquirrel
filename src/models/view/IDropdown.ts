import { IDropdownItem } from './IDropdown';

export interface IDropdownItem {
  name: string;
  code: string;
  onSelect: (name: string, code: string) => void;
  icon: string;
}

export interface IDropdown {
  selectedItem: string;
  items: IDropdownItem[];
  onClick: () => void;
  onSelect: (name: string, code: string) => void;
  show: boolean;
  label: string;
  update: (data: any) => void;
}