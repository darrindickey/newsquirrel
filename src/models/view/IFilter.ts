import { IFilter } from '../data/IFilter';

export interface IFilter {
  name: string;
  value: string;
  selectFilter: (filter: IFilter) => void;
  selected: boolean;
};