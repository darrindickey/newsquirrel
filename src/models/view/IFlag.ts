export interface IFlag {
  code: string;
}