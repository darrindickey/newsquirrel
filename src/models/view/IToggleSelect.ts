import ToggleSelectSize from "../../enums/toggleSelectSize";

export interface IToggleItem {
  name: string;
  value: string;
  onClick: (name: string, value: number) => void;
  selected: boolean;
  size: ToggleSelectSize;
  theme: string;
}

export interface IToggleSelect {
  label: string;
  items: IToggleItem[];
  size: ToggleSelectSize;
  onToggle: (name: string) => void;
  theme: string;
  type?: ToggleType;
  update: (variables: any) => void;
}

export enum ToggleType {
  APP_MODE = 1,
  SORT_ARTICLES = 2,
  RESIZER = 3,
}