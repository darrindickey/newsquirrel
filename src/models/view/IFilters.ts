import { IFilter } from './IFilter';

export interface IFilters {
  items: IFilter[];
  update: (data: any) => void;
}