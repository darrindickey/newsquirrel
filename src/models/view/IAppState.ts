import { AppMode } from './../../enums/appMode';

import NewsStandSize from '../../enums/newsStandSize';
import { IDateFilter } from './../data/IDateFilter';
import { IArticleCard } from './IArticleCard';


interface IThumbnail {
  media: {
    height: number;
    width: number;
  },
  href: string;
}

interface ILogo {
  media: {
    height: number;
    width: number;
  },
  href: string;
}

interface ILinks {
  thumbnail: IThumbnail[];
  logo: ILogo[];
}

export interface IFrameData {
  url: string;
  meta: {
    date: string;
    description: string;
    site: string;
    author: string;
    title: string;
  };
  links: ILinks;
}

export interface INewsState {
  topHeadlines: IArticleCard[];
  newsArticles: IArticleCard[];
  totalResults: number;
  selectedCountry: string;
  searchResultsCount: number;
  headlinesCount: number;
  isAppBusy: boolean;
  sortByTime: ISortBy[];
  activeSortByTime: ISortBy;
  quickViewEnabled: boolean;
  quickViewData: IFrameData | {};
  quickViewUrl: string;
  quickViewLoading: boolean;
  quickViewPosition: {
    x: number;
    y: number;
  }
  pocketConfig: IPocketState;
  searchTerm: string;
}

export interface ICountry {
  name: string;
  value: string;
}

export interface ISortBy {
  name: string;
  value: string;
}

export interface IAppMode{
  name: string;
  value: AppMode;
}

export interface IOptionsState {
  pageSize: number;
  activePage: number;
  defaultCountries: ICountry[];
  defaultCategories: string[];
  filter: {
    categories: string[]
  },
  sortBy: ISortBy[];
  currentlySortingBy: ISortBy;
  searchingFor: string;
  newsStandSize: NewsStandSize;
  dateFilter: IDateFilter;
  detailedPaneOpen: boolean;
  detailedArticleUrl: string;
  defaultAppMode: IAppMode;
  availableAppModes: IAppMode[];
}

export interface IPocketState {
  requestToken: string;
  accessToken: string;
}

export interface IAppState {
  categories: string[];
  news: INewsState;
  options: IOptionsState
}
