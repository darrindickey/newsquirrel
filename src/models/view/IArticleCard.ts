import NewsStandSize from "../../enums/newsStandSize";
import { IAppMode } from './IAppState';

export interface IArticleCard {
  id: string;
  author: string;
  title: string;
  url: string;
  urlToImage: string;
  description: string;
  publishedAt: string;
  source: string;
}

export interface IArticleCardView extends IArticleCard{
  size: NewsStandSize;
  newSize: NewsStandSize;
  imageLoaded: boolean;
  onImageLoaded: () => void;
  showArticle: (url: string) => void;
  checkArticle: (url: string) => void;
  canEmbedInFrame: boolean;
  appMode: IAppMode;
  updateQuickviewUrl: (data: any) => void;
}