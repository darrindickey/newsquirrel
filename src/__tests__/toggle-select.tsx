import "jest-dom/extend-expect";
import * as React from "react";
import { render } from "react-testing-library";
import ToggleSelect from "../components/toggle-select/toggle-select";
import toggleSelectSize from "../enums/toggleSelectSize";
import { IToggleSelect } from "../models/view/IToggleSelect";

test("Render Toggle Select correctly", () => {
  const onToggleCB = jest.fn();
  const props: IToggleSelect = {
    items: [
      {
        name: "item1",
        onToggle: onToggleCB,
        selected: true,
        size: toggleSelectSize.LARGE,
        theme: 'red',
        value: "item1",
      },
      {
        name: "item2",
        onToggle: onToggleCB,
        selected: false,
        size: toggleSelectSize.LARGE,
        theme: 'red',
        value: "item2",
      }
    ],
    label: "Toggle Select Test Label",
    onToggle: onToggleCB,
    size: toggleSelectSize.LARGE,
    theme: "red",
  };
  const { getByTestId } = render(<ToggleSelect {...props} />);
  const node = getByTestId('rt-toggle-select');
  expect(node).toBeInTheDocument();
  expect(node).toMatchSnapshot();
});
