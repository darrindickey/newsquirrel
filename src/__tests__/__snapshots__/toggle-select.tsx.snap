// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Toggle Select correctly 1`] = `
<div
  class="sc-bdVaJa klqPZc"
  data-testid="rt-toggle-select"
  size="1"
>
  <div
    class="sc-bwzfXH hahZpa"
    label="Toggle Select Test Label"
    size="1"
  >
    Toggle Select Test Label
  </div>
  <ul
    class="sc-htpNat hojiQY"
  >
    <li
      class="sc-bxivhb hAVIEP"
      size="1"
      tabindex="0"
    >
      item1
    </li>
    <li
      class="sc-bxivhb ejaHBv"
      size="1"
      tabindex="0"
    >
      item2
    </li>
  </ul>
</div>
`;
